src/Data/List/Properties.agda

```diff
@@ -818,6 +819,13 @@ module _ {P : Pred A p} (P? : Decidable P) where
   ...  | true  = cong (Prod.map (x ∷_) id) (partition-defn xs)
   ...  | false = cong (Prod.map id (x ∷_)) (partition-defn xs)
 
+  length-partition : ∀ xs → (let (ys , zs) = partition P? xs) →
+                     length ys ≤ length xs × length zs ≤ length xs
+  length-partition []       = z≤n , z≤n
+  length-partition (x ∷ xs) with does (P? x) | length-partition xs
```

Comment:
why do you have to `with` on `length-partition`? In fact, why not use `if_then_else` here?

Reply:
We need the `with` on `does (P? x)` for `partition P? (x ∷ xs)` to reduce.
The additional `with` on `length-partition xs` is a standard technique to get
on the good side of the termination checker now that the (buggy) `with`-inliner
has been removed.

Reply:
Thanks for the first answer, it was not obvious to me that things in the type needed to reduce. This is subtle, and we ought to have a better way to do that than `with`. I was afraid that termination was indeed the reason for the other `with`; sad, but oh well.